feature-libs/checkout/base/root/events/checkout-delivery-address-event.listener.ts

```diff
@@ -0,0 +1,91 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import {
+  DeleteUserAddressEvent,
+  EventService,
+  UpdateUserAddressEvent,
+  UserAddressEvent,
+} from '@spartacus/core';
+import { Subscription } from 'rxjs';
+import { filter } from 'rxjs/operators';
+import { CheckoutDeliveryAddressFacade } from '../facade/checkout-delivery-address.facade';
+import {
+  ClearCheckoutDeliveryAddressEvent,
+  DeliveryAddressClearedEvent,
+  DeliveryAddressSetEvent,
+  ResetCheckoutQueryEvent,
+  ResetDeliveryModesEvent,
+} from './checkout.events';
+
+/**
+ * Checkout delivery address event listener.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class CheckoutDeliveryAddressEventListener implements OnDestroy {
+  protected subscriptions = new Subscription();
```

Comment:
we can safely remove the subscription and OnDestroy lifecycle.

These listeners will always be an active listener :P because they're singletons. Same goes for the other listeners.

Reply:
Hmm, but not in the context of ssr ;)